addons/sale/models/product_product.py

```diff
@@ -17,6 +16,34 @@ class ProductProduct(models.Model):
         compute='_compute_product_is_in_sale_order',
         search='_search_product_is_in_sale_order',
     )
+    last_invoice_since = fields.Char(compute="_compute_last_invoice_since")
+
+    @api.depends_context('formatted_display_name', 'partner_id', 'prioritize_for')
+    def _compute_display_name(self):
+        super()._compute_display_name()
+
+        # Add last invoiced date beside the product's name.
+        if self.env.context.get('formatted_display_name') and\
+            self.env.context.get('prioritize_for') == 'sale' and\
+            self.env['product.template']._must_prioritize_invoiced_product():
+            for product in self:
+                if product.last_invoice_date:
+                    product.display_name = f'`{product.last_invoice_since}` **{product.display_name}**'
+
+    @api.depends('last_invoice_date')
+    def _compute_last_invoice_since(self):
+        self.last_invoice_since = False
+        today = fields.Date.today()
+        for product in self:
+            if product.last_invoice_date:
+                days_count = (today - product.last_invoice_date).days
+                if days_count > 365:
+                    day_value_str = self.env._('%(years_count)sy', years_count=(days_count // 365))
+                elif days_count > 30:
+                    day_value_str = self.env._('%(months_count)smo', months_count=(days_count // 30))
+                else:
+                    day_value_str = self.env._('%(days_count)sd', days_count=days_count)
+                product.last_invoice_since = day_value_str
```

Comment:
This could be an helper function to avoid the duplication and just be call in the _compute_display_name (we don't need a field for it)

Reply:
I agree for the helper function but about the field, I made it to be able to display this value in the kanban view (SO product catalog.) Before that it wasn't a field.
My first solution was to compute the value in JS but it was kind of hacky to display it in the kanban record view, and it duplicated the logic (one in the python and one in the JS), I would prefer to keep it as a compute field.